### PR TITLE
Add support for OIDC authentication

### PR DESCRIPTION
**Pull request by helenst**
See also https://github.com/wellcometrust/archivematica/pull/10 and https://github.com/wellcometrust/archivematica-infra/pull/51

For https://github.com/wellcometrust/platform/issues/3261

### CODE REVIEW

File: storage_service/common/backends.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        """Get the signing key by exploring the JWKS endpoint of the OP."""
+
+        # This method is overridden to provide a fix for "alg" potentially not
+        # being present in the response (it is optional in the spec, and not
+        # provided by Azure)
+
+        # The latest version of mozilla-django-oidc provides this fix, but we
+        # cannot currently use it due to being on Django 1.8. Once we are on a
+        # recent Django version and using mozilla_django_oidc>=1.1.0 then we
+        # can discard this override.
```
Comment (alexwlchan) [inline storage_service/common/backends.py:26]: Useful comment. 👍

File: storage_service/storage_service/settings/base.py [hunk 2]
```diff
@@ -428,6 +429,52 @@ def is_true(env_str):
...
+    OIDC_RP_CLIENT_ID = environ.get("OIDC_RP_CLIENT_ID", "")
+    OIDC_RP_CLIENT_SECRET = environ.get("OIDC_RP_CLIENT_SECRET", "")
+
+    OIDC_OP_AUTHORIZATION_ENDPOINT = ""
+    OIDC_OP_TOKEN_ENDPOINT = ""
+    OIDC_OP_USER_ENDPOINT = ""
+    OIDC_OP_JWKS_ENDPOINT = ""
+
+    AZURE_TENANT_ID = environ.get("AZURE_TENANT_ID", "")
+    if AZURE_TENANT_ID:
```
Comment (alexwlchan) [inline storage_service/storage_service/settings/base.py:452]: What if you don't set an AZURE_TENANT_ID? Can you use OpenID functionality?

**Review by alexwlchan — APPROVED** [review]
Looks good to me, with my limited Django knowledge.

File: storage_service/storage_service/settings/base.py (hunk 2 above)
Comment (helenst) [reply storage_service/storage_service/settings/base.py:452]: Good point. I guess it should attempt to read those from the environment instead.

**Pull request merged**